tests/marqo_test.py

```diff
@@ -146,4 +142,39 @@ def warm_request(self, func, *args, **kwargs):
         This solves the occurence of tests failing due to eventual consistency implemented in marqo cloud.
         '''
         for i in range(5):
-            func(*args, **kwargs) 
+            func(*args, **kwargs)
+
+    def create_cloud_index(self, index_name, index_defaults=None, **kwargs):
```

Comment:
we have to be careful here. If the user specifies index_defaults, they would expect an index to be made with those settings. Because we just return if that index exists, the user may start to index that they don't suspect.

At the very least we need to get the index's settings, and to a comparison the index defaults, raising an error if they don't match.

This behaviour then needs to be clearly specified in the docstring